notify-types/src/event.rs

```diff
@@ -309,8 +309,13 @@ pub struct Event {
     /// creation, they should all go in this `Vec`. Otherwise, using the `Tracker` attr may be more
     /// appropriate.
     ///
-    /// The order of the paths is likely to be significant! For example, renames where both ends of
-    /// the name change are known will have the "source" path first, and the "target" path last.
+    /// The order of the paths is likely to be significant! Furthermore, the paths might be part of
+    /// different "groups". For example, renames where both ends of the name change are known will
+    /// have the "source" paths first, followed by the "target" paths. These two sets of paths are
+    /// in distinct groups.
+    ///
+    /// If there are different path groups, the `path_group_split_index` attribute indicates the
+    /// first index with paths in the second group.
     pub paths: Vec<PathBuf>,
```

Comment:
As this is a behavioral change, we probably want some kind of API change so users explicitly decide how to handle multiple paths in one group, or this newly introduced concept of 2 different groups.

I didn't want to break this into two separate vectors, as to not increase the size of the `Event` struct. Maybe we should make `paths` private and have a `fn path_groups(&self) -> (&[PathBuf], Option<&[PathBuf]>)` method?

Reply:
How about instead of having the api on the attrs interact directly with the fields, there's a more intuitive API on Event like `renamed_from_paths() -> Option<&[Path]>` and `renamed_to_paths() -> Option<&[Path]>` that interprets the kind/paths/attr information?

Reply:
I was, yes.